src/threading/MsgThread.cc

```diff
@@ -178,6 +179,29 @@ MsgThread::MsgThread() : BasicThread(), queue_in(this, 0), queue_out(0, this)
 	child_finished = false;
 	failed = false;
 	thread_mgr->AddMsgThread(this);
+
+	// Open a socket pair used to trigger the iosource loop into processing
+	// new messages on this thread. It's set to non-block because we want to
+	// read out all notifications on every pass without blocking.
+	socketpair(AF_UNIX, SOCK_STREAM, 0, ready_pair);
+	int flags = fcntl(ready_pair[0], F_GETFL);
+	if ( flags != -1 )
+		fcntl(ready_pair[0], F_SETFL, flags | O_NONBLOCK);
+
+	flags = fcntl(ready_pair[0], F_GETFL);
```

Comment:
Maybe copy-paste error.
```suggestion
	flags = fcntl(ready_pair[1], F_GETFL);
```

Reply:
Whoops, good catch.

Reply:
Converted the two socketpairs into Flares.